discovery/data_sharing/v0.1/data_sharing.md

```diff
@@ -0,0 +1,297 @@
+# "Fediscovery": Data Sharing Specification
+
+## Overview
+
+Specification identifier: `data_sharing`
+
+In order to be able to present results, discovery providers need to be
+able to request content from connected fediverse servers. Two different
+concerns need to be distinguished: Retrieving new content as it is
+created and retrieving historic content to backfill search indexes.
+
+## Restrictions On Content Being Shared
+
+A single FASP will never be used by all fediverse servers. To ensure it
+still gets a broad view not limited to a small number of servers,
+fediverse servers MUST share not only local but also remote content with
+the FASP.
+
+Since this may result in multiple servers sharing the same content with
+a single FASP, FASPs MUST deduplicate content.
+
+As a single server is not authoritative for all the content shared and
+to save bandwidth only URIs of content are actually exchanged with the
+FASP. The FASP is then responsible for retrieving the content and
+working with it. The exact details of this differ between capabilties
+and are described in the respective sections.
+
+A fediverse server MUST make sure it only shares content that it is
+actually allowed to share. It MUST NOT share any content that is not
+public. In the case of account and post data it MUST make sure to only
+share content where the creator has opted in to discovery.
+
+The FASP in turn MUST also ensure that creators have opted in to
+discovery before storing and indexing content.
```

Comment:
What happens if this value changes? Should we send events to say "hey, this actor is now discoverable" or "that actor is now no longer discoverable" ?

Reply:
I added a paragraph at the end describing the need to periodically refetch the content. As an interval I chose one week as I think this is the interval after which mastodon considers an object to be stale.

Reply:
Yes and this is all part of the spec already, *but* as always the server talking to the FASP might not know about these things if they happen on third-party server. It should, but there are no guarantees.

Hence we will need both: Servers announcing changes to FASP and FASP rechecking content regularly.